aten/src/ATen/native/cuda/EmbeddingBackwardKernel.cu

```diff
@@ -0,0 +1,311 @@
+#include <ATen/ATen.h>
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/TensorUtils.h>
+#include <ATen/NativeFunctions.h>
+
+#include <ATen/AccumulateType.h>
+
+#include <THC/THCDeviceUtils.cuh>
+#include <THC/THCTensorMathReduce.cuh>
+#include <THC/THCTensorSort.cuh>
+#include <THC/THCThrustAllocator.cuh>
+#include <THC/THCAtomics.cuh>
+
+#include <thrust/execution_policy.h>
+#include <thrust/unique.h>
+#include <thrust/device_vector.h>
+
+namespace at {
+namespace native {
+
+namespace {
+
+// The maximum block size in CUDA
+constexpr int MAX_BLOCK_SIZE = 1024;
+// Number of rows each thread should sum initially
+constexpr int NROWS_PER_THREAD = 10;
```

Comment:
how does this number impact the performance?

Reply:
Added some info here: https://github.com/pytorch/pytorch/pull/22016/commits/14a597baca9e0fe47a46a61e8627f5f2859872d2